presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -96,6 +96,8 @@
 
     private List<String> resourceConfigFiles = ImmutableList.of();
 
+    private DataSize maxLineLength = new DataSize(100, DataSize.Unit.MEGABYTE);
```

Comment:
@findepi, do you have a suggestion for a good default here?

Reply:
@dain Piotr had this comment before:
findepi 13 hours ago  Member
Once this becomes configurable, I am not that much concerned about the value.. The only problem would be that in case of a too long line, exception message would not be helpful in tracing the config switch governing the behavior.
As this is supposed to be OOM-preventing safety mechanism only, we can start with 100M being the default and revisit the decision if we face any problems with this.

Reply:
What does the error message look like when we hit this?

Reply:
@dain It's just this line in GC log:
java.lang.OutOfMemoryError: Requested array size exceeds VM limit

However after this change there shouldn't be hitting this anymore.

Reply:
Right.  I mean when we hit this limit do we get something like "Test line larger than 100MB"?

Reply:
@dain No, there is no other line. This exception/error is not caught and rethrown by Hadoop or Presto therefore there isn't such information. It seems the correct fix is for Hadoop to honor the VM limit on array size and forbid allocating array of size beyond the limit, but it's hadoop code and is out of our hands. For now, if we don't enable this max line length configuration change and hit this error again, we could catch it in com.facebook.presto.hive.HiveUtil.createRecordReader() and rethrow with some useful information. Do we want to do that?

Reply:
@dain I think I misunderstood your question in my previous reply. After this change there won't be such errors. According to the comment in code the maxBytesToConsume is just a hint and it's ok for the line length to be above it:
/**
   * Read one line from the InputStream into the given Text.
   *
   * @param str the object to store the given line (without newline)
   * @param maxLineLength the maximum number of bytes to store into str;
   *  the rest of the line is silently discarded.
   * @param maxBytesToConsume the maximum number of bytes to consume
   *  in this call.  This is only a hint, because if the line cross
   *  this threshold, we allow it to happen.  It can overshoot
   *  potentially by as much as one buffer length.
   *
   * @return the number of bytes read including the (longest) newline
   * found.
   *
   * @throws IOException if the underlying stream throws
   */
  public int readLine(Text str, int maxLineLength,
                      int maxBytesToConsume) throws IOException

Reply:
Wow. I would not have expected that.  @findepi does that seem reasonable?

Reply:
With David's suggest config changes, I think this is reasonable.

Reply:
Cool. As long as we agree that truncating the line at 100MB is fine.

Reply:
> the rest of the line is silently discarded.

a bit scary. Is it really discarded or is it taken to constitute a next line?

Reply:
@findepi The chars after the limit is really discarded.